yeps/yep-314.md

```diff
@@ -0,0 +1,48 @@
+---
+yep: 314
+title: supports-collect-measured
+author: Blaise Thompson <blaise@untzag.com>
+status: draft
+tags: trait
+post-history: 2023-09-28
+---
+
+# Abstract
+
+This YEP introduces a message "collect_measured", a way to grab multiple measurements from a sensor simultaniously.
+
+# Table of Contents
+
+[TOC]
+
+# Motivation
+
+Some experiments incorporate sensors that are very fast or very slow compared to other pieces of hardware being controlled. In such cases it might be natural to let the sensor run asyncronously with other hardware, simply recording each time there is a new sensor reading. Unfortunately, the design of the "get_measured" message defined by is-sensor does not make asyncronus acquistion easy. Clients must ensure that they are polling quickly in order to ensure they don't miss measurements. In extreme cases, sensors are so fast that client polling is simply not practical.
```

Comment:
Just to make sure I am understanding the application, this is most useful for when sensors are very fast?  Does this trait also offer advantages when sensors are very slow?

Reply:
I agree it's most useful for very fast things.

For very slow things it's also great, because you just don't have to poll.

It's useful for any measurement that's fundamentally asynchronous with your central datastream.